catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/ResultDAGConverter.java

```diff
@@ -2617,4 +2700,44 @@ private static String getCbrnAlarmClassification(Serializable value) {
 
     return cbrnAlarmClassification;
   }
+
+  private static String translateType(Metacard metacard) {
+    String type = "Text";
+    Attribute attribute = metacard.getAttribute(Core.DATATYPE);
+    if (attribute != null) {
+      type = String.valueOf(attribute.getValue());
+    }
+
+    String result = typeConversionMap.get(type);
+
+    // If result is "DOCUMENT", try to refine based on mime type or attributes
+    String mimeType = "";
+    if (StringUtils.isNotBlank(result) && NsiliProductType.DOCUMENT.getSpecName().equals(result)) {
+      if (metacard.getAttribute(Isr.CHEMICAL_BIOLOGICAL_RADIOLOGICAL_NUCLEAR_CATEGORY) != null) {
+        // CBRN message must have a mandatory category - so this should be present
+        result = NsiliProductType.CBRN.getSpecName();
+      } else { // try by the mime type
+        attribute = metacard.getAttribute(Media.TYPE);
+        if (attribute != null) {
+          mimeType = String.valueOf(attribute.getValue());
+        }
+
+        if (StringUtils.equals("text/plain", mimeType)) {
+          result = NsiliProductType.MESSAGE.getSpecName();
+        }
+        /* Ignoring the report type for now - just leave text/xml a document
+        } else if (StringUtils.equals("text/xml", mimeType)) {
+          result = NsiliProductType.REPORT.getSpecName();
+        }
+        */
```

Comment:
Should we just remove this code?

Reply:
That would be fine. This would be part of the ticket to evaluate XML files and discover their underlying NSIL type.